TEAMMATES/teammates
# [#9536] Feedback Submit Page E2E Test

Part of #9536

Outline of Solution

- Add methods to BaseE2ETestCase to compare emails sent
- Add FeedbackSubmitPage to represent feedback submission page
- Add registration key to result of getStudentAction for backdoor requests.
- Add FeedbackSubmitPageE2ETest to test general actions on the submit page:
    - verify loaded session data
    - questions with giver type instructor
    - questions with giver type students
    - verify recipients: students
    - verify recipients: instructors
    - verify recipients: team mates
    - verify recipients: teams
    - submit partial response
    - cannot submit in closed session
    - can submit in grace period
    - confirmation email
    - add comment
    - edit comment
    - delete comment

Responding to each different specific question type will be tested in their own individual tests instead of here.
